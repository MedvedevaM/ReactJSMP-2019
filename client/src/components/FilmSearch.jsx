import React from 'react';
import './../css/font-awesome/css/font-awesome.min.css';

export const FilmSearch = (props) => {
    const { chosenFilm, onSearchModeClick, onSearchParameterButtonClick, searchBy } = props;
    if (chosenFilm) {
        const release_year = chosenFilm.release_date ? chosenFilm.release_date.slice(0, 4) : null;
        return (
            <header id='home' className="header-bg">
                <div className="transparent-black-header-bg">
                    <div className="container flex space-between">
                        <a href='#home' className='logo'>netflixroulette</a>
                        <button onClick={onSearchModeClick} className='search-button-default-mode bold'>Search</button>
                    </div>
                    <div className='chosen-film flex'>
                        <img src={chosenFilm.poster_path} />
                        <section>
                            <div className='flex'>
                                <h1>{chosenFilm.title}</h1>
                                <p className='rating'>{chosenFilm.vote_average}</p>
                            </div>
                            <p>{chosenFilm.tagline}</p>
                            <div className='flex'>
                                <p className='bold'>{release_year}</p>
                                <p className='bold'>{chosenFilm.runtime ? `${chosenFilm.runtime} min` : null}</p>
                            </div>
                            <p>{chosenFilm.overview}</p>
                        </section>
                    </div>
                </div>
            </header>
        )
    }

    const chosenParameter = 'search-parameters uppercase bold chosen-parameter';
    const notChosenParameter = 'search-parameters uppercase bold';
    return (
        <header id='home' className="header-bg">
            <div className="transparent-black-header-bg">
                <div className="container">
                    <a href='#home' className='logo'>netflixroulette</a>
                    <h1>Find your movie</h1>
                    <input className='search-input' type='text' value={props.value} onChange={props.searchFilms}></input>
                    <i className="fa fa-search search-input-enter" aria-hidden="true"></i>
                    <div className='flex search-parameters' onClick={onSearchParameterButtonClick}>
                        <p>Search by</p>
                        <button id='Title' className={searchBy === 'Title' ? chosenParameter : notChosenParameter}>Title</button>
                        <button id='Genre' className={searchBy === 'Genre' ? chosenParameter : notChosenParameter}>Genre</button>
                    </div>
                </div>
            </div>
        </header>
    )
};