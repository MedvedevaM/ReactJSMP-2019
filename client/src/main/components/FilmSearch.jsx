import React from "react";
import { RadioButtons } from "./RadioButtons.jsx";

export const FilmSearch = (props) => {
    const { chosenFilm, onSearchModeClick, onSearchParameterClick, searchBy } = props;
    if (chosenFilm) {
        const release_year = chosenFilm.release_date ? chosenFilm.release_date.slice(0, 4) : null;
        return (
            <header id="home" className="header-bg">
                <div className="transparent-black-header-bg">
                    <div className="container flex space-between">
                        <a href="#home" className="logo common-color-1">netflixroulette</a>
                        <button onClick={onSearchModeClick} className="search-button-default-mode common-color-1 bold">Search</button>
                    </div>
                    <div className="chosen-film flex">
                        <img src={chosenFilm.poster_path} alt={chosenFilm.title} />
                        <section>
                            <div className="flex">
                                <h1>{chosenFilm.title}</h1>
                                <p className="rating">{chosenFilm.vote_average}</p>
                            </div>
                            <p>{chosenFilm.tagline}</p>
                            <div className="flex">
                                <p className="bold">{release_year}</p>
                                <p className="bold">{chosenFilm.runtime ? `${chosenFilm.runtime} min` : null}</p>
                            </div>
                            <p>{chosenFilm.overview}</p>
                        </section>
                    </div>
                </div>
            </header>
        )
    }

    const searchParameters = ["Title", "Genre"];
    return (
      <header id="home" className="header-bg">
        <div className="transparent-black-header-bg">
          <div className="container">
            <a href="#home" className="logo common-color-1">
              netflixroulette
            </a>
            <h1>Find your movie</h1>
            <input className="search-input" type="text" value={props.value} onChange={props.searchFilms} />
            <i className="fa fa-search search-input-enter common-color-1" aria-hidden="true"/>
            <div className="flex search-parameters">
                <p>Search by</p>
                <RadioButtons onClick={onSearchParameterClick} searchBy={searchBy} parameters={searchParameters}/>
            </div>
          </div>
        </div>
      </header>
    );
};