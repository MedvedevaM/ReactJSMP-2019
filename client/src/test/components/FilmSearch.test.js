import React from "react";
import { shallow, configure } from "enzyme";
import { FilmSearch } from "../../main/components/FilmSearch.jsx";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const chosenFilm = {
    release_date: "2020-05-01",
    title: "Guardians of the Galaxy Vol. 3",
    vote_average: 8,
    runtime: 100,
    tagline: "Tag line",
    overview: "The third film based on Marvel's Guardians of the Galaxy.",
    poster_path: "https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg"
};

it("renders correctly", () => {
    const filmSearch = shallow(<FilmSearch />);
    expect(filmSearch).toMatchSnapshot();
});
it("Film search with chosen film", () => {
    const filmSearch = shallow(<FilmSearch chosenFilm={chosenFilm} />);
    expect(filmSearch.contains(
        <div className="container flex space-between">
            <a href="#home" className="logo common-color-1">netflixroulette</a>
            <button className="search-button-default-mode common-color-1 bold">Search</button>
        </div>
        )).toBe(true);

    expect(filmSearch.contains(
        <div className="chosen-film flex">
            <img src="https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg" alt="Guardians of the Galaxy Vol. 3" />
            <section>
                <div className="flex">
                    <h1>Guardians of the Galaxy Vol. 3</h1>
                    <p className="rating">{chosenFilm.vote_average}</p>
                </div>
                <p>Tag line</p>
                <div className="flex">
                    <p className="bold">2020</p>
                    <p className="bold">100 min</p>
                </div>
                <p>The third film based on Marvel's Guardians of the Galaxy.</p>
            </section>
        </div>)).toBe(true);
});

