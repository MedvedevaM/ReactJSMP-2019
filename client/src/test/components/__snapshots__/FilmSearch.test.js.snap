// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`film search rendering renders correctly 1`] = `
<header
  className="header-bg"
  id="home"
>
  <div
    className="transparent-black-header-bg"
  >
    <div
      className="container"
    >
      <a
        className="logo common-color-1"
        href="#home"
      >
        netflixroulette
      </a>
      <h1>
        Find your movie
      </h1>
      <input
        className="search-input"
        type="text"
      />
      <i
        aria-hidden="true"
        className="fa fa-search search-input-enter common-color-1"
      />
      <div
        className="flex search-parameters"
      >
        <p>
          Search by
        </p>
        <RadioButtons
          parameters={
            Array [
              "Title",
              "Genre",
            ]
          }
        />
      </div>
    </div>
  </div>
</header>
`;
